### PR TITLE
Add ability to filter the source URL in the references

### PR DESCRIPTION
**Pull request by akirk**
This allows to create a filter that modifies the URLs for the files listed under References.

Example usage:

``` php
add_filter( 'gp_reference_source_url', function( $source_url, $project, $file, $line ) {
// Libraries don't reside in our repo but we have them translatable through GlotPress.
if ( preg_match( '#^lib/#', $file ) ) {
     return false; 
}
return $source_url;
}, 10, 4 );
```

Possible uses:
- for projects that include strings from different sources
- for only partially linkable source texts

This fixes a `TODO` source line.

I have already used `@since 2.2.0` for the docs but this can of course be changed depending on which version this lands with.

### CODE REVIEW

**Comment by ocean90** [comment]
Why is the filter in the template and not in `GP_Project::source_url()`?

**Comment by akirk** [comment]
@ocean90 Fair point. I've moved it. I didn't think much to put it somewhere else because of the TODO comment.

File: gp-includes/things/project.php [hunk 1]
```diff
@@ -233,10 +233,25 @@ public function regenerate_paths( $parent_project_id = null ) {
...
+		 * Allows per-reference overriding of the source URL defined as project setting.
+		 *
+		 * @since 2.2.0
+		 *
+		 * @param string        $source_url   The originally generated source URL.
+		 * @param GP_Project    $project      The current project.
+		 * @param string        $file         The referenced file name.
+		 * @param string        $line         The line number in the referenced file.
+		 *
+		 * @return string|false
```
Comment (ocean90) [inline gp-includes/things/project.php]: We don't use `@return` tags for hook docs.

File: gp-includes/things/project.php [hunk 2]
```diff
@@ -233,10 +233,25 @@ public function regenerate_paths( $parent_project_id = null ) {
...
+
+		/**
+		 * Allows per-reference overriding of the source URL defined as project setting.
+		 *
+		 * @since 2.2.0
+		 *
+		 * @param string        $source_url   The originally generated source URL.
+		 * @param GP_Project    $project      The current project.
+		 * @param string        $file         The referenced file name.
+		 * @param string        $line         The line number in the referenced file.
```
Comment (ocean90) [inline gp-includes/things/project.php]: There should be only one space between `GP_Project` and `$project`. Same for `$source_url` and its descriptions. The other parameters should be aligned accordingly.

`$source_url` can be false too:
`@param string|false $source_url The originally generated source URL, or false if no URL is available.`

**Pull request merged**